java-squid/src/main/java/org/sonar/java/model/statement/ForEachStatementImpl.java

```diff
@@ -37,14 +35,23 @@
 import java.util.Iterator;
 
 public class ForEachStatementImpl extends JavaTree implements ForEachStatement {
+  private final InternalSyntaxToken forKeyword;
+  private final InternalSyntaxToken openParenToken;
   private final VariableTree variable;
+  private final InternalSyntaxToken colonToken;
   private final ExpressionTree expression;
+  private final InternalSyntaxToken closeParenToken;
   private final StatementTree statement;
 
-  public ForEachStatementImpl(VariableTree variable, ExpressionTree expression, StatementTree statement, AstNode... children) {
+  public ForEachStatementImpl(InternalSyntaxToken forKeyword, InternalSyntaxToken openParenToken, VariableTree variable, InternalSyntaxToken colonToken,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)